gateway/src/function.rs

```diff
@@ -124,6 +125,13 @@ impl FunctionConfig {
         }
     }
 
+    #[instrument(skip(
+        content_blocks,
+        usage,
+        model_inference_results,
+        inference_config,
+        inference_params
+    ))]
```

Comment:
You're leaking stuff here (e.g. `self`). `skip_all` + explicitly select what you want.

I haven't tested but think the following should work.

```
#[instrument(skip_all, fields(inference_id))]
```

Reply:
that works!